pkg/utils/nodes.go

```diff
@@ -0,0 +1,49 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var logger = ctrl.Log.WithName("utils-nodes")
+
+// similar to getNodes - https://github.com/medik8s/node-maintenance-operator/blob/main/test/e2e/node_maintenance_test.go#L258-L282
+// getNodeNames returns a list of node names or an error it can't get the node names
+func getNodeNames(r client.Reader) ([]string, error) {
+	nodes := make([]string, 0)
+	nodesList := &corev1.NodeList{}
+	err := r.List(context.TODO(), nodesList, &client.ListOptions{})
+	//wait one second
+	time.Sleep(1 * time.Second)
+	if err != nil {
+		logger.Error(err, "failed to get node names")
+		return nil, err
+	}
+	for _, node := range nodesList.Items {
+		if node.Labels == nil {
+			logger.Info("node has no role label", "Node", node.Name)
+			continue
+		}
+		nodes = append(nodes, node.Name)
+	}
+	return nodes, nil
+}
+
+// isNodeNameValid returns an error if nodeName doesn't match the node names from getNodeNames, otherwise a nil
+func IsNodeNameValid(r client.Reader, nodeName string) error {
+	clusterNodes, err := getNodeNames(r)
+	if err != nil {
+		return err
+	}
+	for _, node := range clusterNodes {
```

Comment:
Have one function that go over the list and search for a specific given name or to instead of loop over the list of node names I should search differently?

Reply:
No. Do not get a node list at all. Just do a `get` for a single node with the given name. When it exists -> valid, if not -> not valid 🙂

Reply:
More reasons for getting the `get` the verb ;)